TypeError: cannot use a str to initialize an array with typecode 'B'

When running a ported version of multimaster_fkie for Python 3, the following error occurs:

```
[ERROR] [1587930573.923334]: Traceback (most recent call last):
  File "<string>", line 109, in main
  File "/home/hermann/turag/programmierung/ros/src/multimaster_fkie/master_discovery_fkie/src/master_discovery_fkie/master_discovery.py", line 581, in __init__
    local_addr = roslib.network.get_local_address()
  File "/opt/ros/melodic/lib/python3.8/site-packages/roslib/network.py", line 139, in get_local_address
    addrs = get_local_addresses()
  File "/opt/ros/melodic/lib/python3.8/site-packages/roslib/network.py", line 187, in get_local_addresses
    buff = array.array('B', '\0' * max_bytes)
TypeError: cannot use a str to initialize an array with typecode 'B'
```

Affected file: https://github.com/ros/ros/blob/noetic-devel/core/roslib/src/roslib/network.py#L197

--- a/core/roslib/src/roslib/network.py
+++ b/core/roslib/src/roslib/network.py
@@ -194,7 +194,7 @@ def get_local_addresses():
 
         max_bytes = 32 * ifsize
         # according to http://docs.python.org/library/fcntl.html, the buffer limit is 1024 bytes
-        buff = array.array('B', '\0' * max_bytes)
+        buff = array.array('B', b'\0' * max_bytes)
         # serialize the buffer length and address to ioctl
         sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         info = fcntl.ioctl(sock.fileno(), SIOCGIFCONF,

